File: 프로그래머스/0/120878. 유한소수 판별하기/유한소수 판별하기.py
import math
def solution(a, b):
    answer = 0
    g = math.gcd(a, b)
    a = a // g
    b = b // g
    s = [2, 5]
    while b != 1 :
        for i in range(2, b + 1) :
            if b % i == 0 :
                if i not in s :
                    return 2
                else :
                    b = b // i
                    break

    return 1
    
    return answer

File: 프로그래머스/0/120878. 유한소수 판별하기/유한소수 판별하기.java
class Solution {
    public static int solution(int a, int b) {
        int gcd = 0;
        if (a >= b) {
            gcd = gcd(a, b);
        } else {
            gcd = gcd(b, a);
        }

        b = b / gcd;
        int count = 0;
        while (b != 1) {
            if (b % 2 == 0) {
                b /= 2;
                count = 0;
            }
            if (b % 5 == 0) {
                b /= 5;
                count = 0;
            }
            count++;

            if (count >= 2) {
                return 2;
            }
        }
        return 1;
    }

    public static int gcd(int a, int b) {
    if (b == 0) {
        return a;
    }
    return gcd(b, a % b);
    }
}

